### PR TITLE
Carbon offset support

### PR DESCRIPTION
**Pull request by nwithan8**
# Description

- New functions to create, buy and re-rate a shipment with a carbon offset
- New unit tests
- Fix function using deprecated function

# Testing

- New unit tests

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement (fixing a typo, updating readme, renaming a variable name, etc.)

### CODE REVIEW

File: rate.go [hunk 1]
```diff
@@ -7,26 +7,27 @@ import (
...
+	Rate                   string        `json:"rate,omitempty"`
+	Currency               string        `json:"currency,omitempty"`
+	RetailRate             string        `json:"retail_rate,omitempty"`
+	RetailCurrency         string        `json:"retail_currency,omitempty"`
+	ListRate               string        `json:"list_rate,omitempty"`
+	ListCurrency           string        `json:"list_currency,omitempty"`
+	DeliveryDays           int           `json:"delivery_days,omitempty"`
+	DeliveryDate           *time.Time    `json:"delivery_date,omitempty"`
+	DeliveryDateGuaranteed bool          `json:"delivery_date_guaranteed,omitempty"`
+	EstDeliveryDays        int           `json:"est_delivery_days,omitempty"`
```
Comment (Justintime50) [inline rate.go:28]: What changed between these lines? Tabs to spaces? They shouldn't have been altered... Also gofmt should have complained about this if they changed indentation strategies.

File: tests/carbon_offset_test.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	shipment, err := client.CreateShipment(c.fixture.OneCallBuyCarbonOffsetShipment())
+	require.NoError(err)
+
+	rates, err := client.RerateShipment(shipment.ID)
+	require.NoError(err)
+
+	assert.Equal(reflect.TypeOf([]*easypost.Rate{}), reflect.TypeOf(rates))
+	for _, rate := range rates {
+		assert.Equal(reflect.TypeOf(&easypost.Rate{}), reflect.TypeOf(rate))
+	}
```
Comment (Justintime50) [inline tests/carbon_offset_test.go:60]: We aren't testing carbon offset at all here. Please have this match the Java assertions for this test.

File: rate.go (hunk 1 above)
Comment (nwithan8) [reply rate.go:28]: I believe it was the indention (and, obviously, the new CarbonOffset property)

File: tests/carbon_offset_test.go (hunk 2 above)
Comment (nwithan8) [reply tests/carbon_offset_test.go:60]: Good catch

File: rate.go (hunk 1 above)
Comment (Justintime50) [reply rate.go:28]: Right, the indentation should remain the same while the CarbonOffset should be the only alteration here.

File: tests/carbon_offset_test.go (hunk 2 above)
Comment (nwithan8) [reply tests/carbon_offset_test.go]: Copy-pasta resulted in the wrong thing getting deleted. Resolved.

**Review by Justintime50 — APPROVED** [review]

**Review by Justintime50 — APPROVED** [review]

**Pull request merged**